### PR TITLE
Cors settings

### PR DESCRIPTION
**Pull request by krishnabvkr**
Reject CORS requests from non whitelisted origins
Log rejected domains in access.log
Unit tests added.

### CODE REVIEW

**Review by malibuworkcrew — APPROVED** [review]

File: src/main/scala/com/webtrends/harness/component/akkahttp/routes/RouteGenerator.scala [hunk 1]
```diff
@@ -66,22 +67,22 @@ object RouteGenerator {
...
-        corsSupport(method, enableCors) {
-          httpMethod(method) {
-            parameterMap { paramMap: Map[String, String] =>
-              extractRequest { request =>
-                val reqHeaders = request.headers.map(h => h.name.toLowerCase -> h.value).toMap
-                val httpEntity = getPayload(method, request)
-                val locales = requestLocales(reqHeaders)
-                val reqWrapper = AkkaHttpRequest(path, paramHoldersToList(segments), request.method, request.protocol,
-                  reqHeaders, paramMap, System.currentTimeMillis(), locales, httpEntity)
+        httpMethod(method) {
```
Comment (Kraagen) [inline src/main/scala/com/webtrends/harness/component/akkahttp/routes/RouteGenerator.scala]: Are we sure we'll still support pre-flight requests if this is before the corsSupport directive?

We also need to make sure the preflight requests work with multiple endpoints on the same path with different methods.

Comment (krishnabvkr) [reply src/main/scala/com/webtrends/harness/component/akkahttp/routes/RouteGenerator.scala]: No, pre-flight requests are not working when httpMethod is before the corsSupport. I have updated the code by placing httpMethod after corsSupport, now pre-flight requests are working.
Currently the pre-flight requests with multiple endpoints on the same path with different methods are not working as expected it returns only latest registered method of that endpoint, we should work on this issue. I was the one who confirmed previously it was working fine but I was tested in wrong way, sorry for that.

File: src/main/scala/com/webtrends/harness/component/akkahttp/routes/RouteGenerator.scala [hunk 2]
```diff
@@ -209,14 +212,21 @@ object RouteGenerator {
...
         .map(language => Locale.forLanguageTag(language.getRange)).toList
       case _ => Nil
     }
 
+  private def corsRejectionHandler(request: AkkaHttpRequest, accessLogIdGetter:Option[AkkaHttpRequest => String]): RejectionHandler =
+    RejectionHandler
+      .newBuilder()
+      .handleAll[javadsl.CorsRejection] { rejections =>
+        val causes = rejections.map(_.cause.description).mkString(", ")
+        accessLogIdGetter.foreach(g => AccessLog.logAccess(request, g(request), StatusCodes.Forbidden))
```
Comment (Kraagen) [inline src/main/scala/com/webtrends/harness/component/akkahttp/routes/RouteGenerator.scala:227]: We still want to write to the access log even if accessLogIdGetter is None.

Comment (AlexanderNull) [reply src/main/scala/com/webtrends/harness/component/akkahttp/routes/RouteGenerator.scala:227]: At this point in the code if accessLogIdGetter is None then access logging has not been enabled for this service.

Comment (AlexanderNull) [reply src/main/scala/com/webtrends/harness/component/akkahttp/routes/RouteGenerator.scala:227]: See line https://github.com/oracle/wookiee-akka-http/blob/2.0.x/src/main/scala/com/webtrends/harness/component/akkahttp/routes/AkkaHttpEndpointRegistration.scala#L56

**Review by AlexanderNull — APPROVED** [review]

**Review by Kraagen — APPROVED** [review]

**Pull request merged**